docusaurus/docs/extensions/extensions-configuration.md

```diff
@@ -0,0 +1,956 @@
+  path: `/${ YOUR_EXT_PRODUCT_NAME }/c/:cluster`,
+  params: {
+    cluster: '_',
+    product: YOUR_EXT_PRODUCT_NAME
+  }
+}
+```
+
+As we can see we have dismissed the `product` wildcard on the `path` and replaced it with the Extension product name to make it unique. With the `product` param we make sure that the is taken to the correct product at all time.
+This structure on the above example ensures that all the wiring needed for the Extension to work properly on Rancher Dashboard is done. There's even the case where the wildcard `resource` needs to be defined in order to display information about Kubernetes resources or custom CRDs. An example of a resource route in a top-level Extension product would be:
+
+```ts
+const YOUR_EXT_PRODUCT_NAME = 'myExtension';
+const RESOURCE_NAME = 'my-resource-name';
+
+const routeForATopLevelProductResource = {
+  name: `${ YOUR_EXT_PRODUCT_NAME }-c-cluster-resource`,
+  path: `/${ YOUR_EXT_PRODUCT_NAME }/c/:cluster/:resource`,
+  params: {
+    cluster: '_',
+    product: YOUR_EXT_PRODUCT_NAME
+    resource: RESOURCE_NAME
+  }
+}
+```
+
+With this quick guide on routing for top-level Extension products, we should be ready to cover the registration of custom pages, resource pages and general route definition. 
+
+
+## Defining a custom page for an Extension (virtualType)
+As we've seen from the previous example, a developer can register a top-level product with the `product` function. How about adding a custom page to your extension product? To do that, we can use the function `virtualType` coming from `$plugin.DSL`. As an example usage of that method, one could do the following:
+
+```ts
+// this is the definition of a "blank cluster" for Rancher Dashboard
+import { BLANK_CLUSTER } from '@shell/store';
+
+export function init($plugin, store) {
+  const YOUR_PRODUCT_NAME = 'myProductName';
+  const CUSTOM_PAGE_NAME = 'page1';
+  
+  const { 
+    product,
+    virtualType
+  } = $plugin.DSL(store, YOUR_PRODUCT_NAME);
+
+  // registering a top-level product
+  product({
+    icon: 'gear',
+    inStore: 'management',
+    weight: 100,
+    to: { // this is the entry route for the Extension product, which is registered below
+      name: `${ YOUR_PRODUCT_NAME }-c-cluster-${ CUSTOM_PAGE_NAME }`,
+      params: {
+        product: YOUR_PRODUCT_NAME,
+        cluster: BLANK_CLUSTER
+      }
+    }
+  });
+
+  // => => => creating a custom page
+  virtualType({
+    labelKey: 'some.translation.key',
+    name:     CUSTOM_PAGE_NAME,
+    route:    {
+      name:   `${ PRODUCT_NAME }-c-cluster-${ CUSTOM_PAGE_NAME }`,
+      params: {
+        product: YOUR_PRODUCT_NAME,
+        cluster: BLANK_CLUSTER
+      }
+    }
+  });
+}
+```
+
+With the route definition in the router (check the [Extension Routing](#extension-routing-definition) chapter), you can define which Vue component will be loaded as a custom page. That will act as a "blank canvas" to render anything you want.
```

Comment:
`#extension-routing-definition` this chapter doesn't seem to exist. Should this be `#routes-definition-for-an-extension-as-a-top-level-product`.

Reply:
true, although usually the build fails when finding broken links. Fixed